app/views/generic_files/_descriptions.html.erb

```diff
@@ -10,7 +10,7 @@
 
       </div><!-- /well -->
     <div>
-      <%= f.submit class: 'btn btn-primary', onclick: "confirmation_needed = false;", id: "upload_submit", name: "update_descriptions" do %>
+      <%= button_tag type: 'submit', class: 'btn btn-primary btn-lg', onclick: "confirmation_needed = false;", id: "upload_submit", name: "update_descriptions" do %>
```

Comment:
You should be able to omit the `type: 'submit'` as that should be the default. Also, is confirmation_needed the same as the `confirm:` option? Otherwise this is fine as-is. Just my picky comments.